### DISCUSSION TITLE
How to apply a filter to a func.group_concat column?

### DISCUSSION THREAD

**Discussion by fazal-kh4n** [sqlalchemy/sqlalchemy#7264] [Usage Questions] [answered]
I have the following sort of initial query:

```
query = session.query(
        Project.project_id,
        Project.project_name,
        Project.forecast_client_id,
        func.group_concat(Channel.channel_name.op('separator')('__')).label('project_channels'),
        .outerjoin(ClientId, ClientId.client_id == Project.client_id) \
        .group_by(Project.project_id)
)
```

Now for a whole bunch of reasons, the above is my starting query. I cannot rework it without massively impacting other parts of the code. Assuming it is a given, how could I filter on the concatenated column?

I tried something akin to the following:

`query.filter(query.project_channels == 'target_channel')`

But this failed.

However doing something like this works:

```
for x in query.all():
    print(x.project_channels)
```

Is there a way I can the first filter somehow? Any help is much appreciated.

Thanks!

**Comment by CaselIT** [comment, marked as answer]
Hi,

There are many alternatives. Below are a few
You can repeat the function in the order by or assign it to a variable and use it both in the column list and in the order by.
Another alternative is to [`column`](https://docs.sqlalchemy.org/en/14/core/sqlelement.html#sqlalchemy.sql.expression.column) with the label name.
A further alternative may be to replace query with `select` and access it using the [`selected_columns`](https://docs.sqlalchemy.org/en/14/core/selectable.html#sqlalchemy.sql.expression.SelectBase.selected_columns) property.

**Reply by fazal-kh4n** [reply]
Thanks @CaselIT ,

I resolved it using a subquery approach, but this is helpful too!
